src/components/LocationSelector/LocationSelector.tsx

```diff
@@ -129,20 +129,37 @@ export const LocationSelector: React.FC<Props> = (props) => {
 
     const [isOpen, setIsOpen] = React.useState(false);
     const [isWrapperFocused, setIsWrapperFocused] = React.useState(false);
+    const [isBrowserTabVisible, setIsBrowserTabVisible] = React.useState(true);
     const buttonRef = React.useRef<HTMLButtonElement>();
 
     const hasLocationsSelected = selectedLocations && selectedLocations.length > 0;
 
+    React.useEffect(() => {
+        const handleFocushandleVisibilityChange = () => {
+            if (document.hidden) {
+                setIsBrowserTabVisible(false);
+            } else {
+                setTimeout(() => setIsBrowserTabVisible(true), 250);
+            }
+        };
+
+        document.addEventListener('visibilitychange', handleFocushandleVisibilityChange);
+
+        return () => {
+            document.removeEventListener('visibilitychange', handleFocushandleVisibilityChange);
+        };
+    });
+
```

Comment:
I noticed that code is duplicated in `SelectBase` component. Could we create some general wrapper for managing the tab's switching behavior? or move it to the `hooks` folder and reuse it after?

Reply:
making a hook out of that is a nice idea, thanks